pkg/detection/detection.go

```diff
@@ -0,0 +1,163 @@
+package detection
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+	"unicode"
+	"unsafe"
+
+	"github.com/buger/jsonparser"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/grafana/loki/v3/pkg/loghttp/push"
+	"github.com/grafana/loki/v3/pkg/logproto"
+	"github.com/grafana/loki/v3/pkg/logql/log/logfmt"
+)
+
+const (
+	AggregatedMetricLabel = "__aggregated_metric__"
```

Comment:
I'm open to other names. Maybe `__rollup__`? or `__service_rollup__` in case we want to add other rollups in the future?